### PR TITLE
[Filebeat] Add mqtt support in Zeek Module

### PR DESCRIPTION
**Pull request by garanews**
added mqtt support

Co-Authored-By: Arcuri Davide <dadokkio@gmail.com>

### CODE REVIEW

**Comment by exekias** [comment]
Thank you so much for contributing! I noticed that the filesets are created inside the `zeek` module. Is that intended? I would expect this to be a new module?

**Comment by exekias** [comment]
Ah sorry, now I understand these are to parse zeek logs on mqtt, sorry for the noise! also pinging @elastic/siem

**Comment by garanews** [comment]
Yes probably better to change title to avoid confusion.
This parses the mqtt log files generated by zeek:
mqtt_connect.log
mqtt_subscribe.log
mqtt_publish.log

**Comment by garanews** [comment]
@andresrc ,  @odacremolbap do you need other info?

**Comment by garanews** [comment]
someone can approve?

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
I pushed some updates to the filesets and rebased it on master. This is ready for review.

**Comment by garanews** [comment]
Thanks

File: x-pack/filebeat/module/zeek/mqtt_connect/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    - name: client_id
+      type: keyword
+      description: >
+        Unique identifier for the client.
+
+    - name: connect_status
+      type: keyword
+      description: >
+        Status message from the server in response to the connect request.
```
Comment (leehinman) [inline x-pack/filebeat/module/zeek/mqtt_connect/_meta/fields.yml:20]: Looking at https://docs.zeek.org/en/master/scripts/policy/protocols/mqtt/main.zeek.html, it looks like MQTT::ConnectInfo can have some other fields not listed like will_topic & will_payload.  I think we should add those so we don't have a default mapping (if it is possible for them to appear in the logs).

File: x-pack/filebeat/module/zeek/mqtt_publish/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    - name: payload
+      type: keyword
+      description: >
+        Payload of the message.
+
+    - name: payload_len
+      type: long
+      description: >
+        The actual length of the payload in the case the payload.
```
Comment (leehinman) [inline x-pack/filebeat/module/zeek/mqtt_publish/_meta/fields.yml:41]: Looking at MQTT::PublishInfo at https://docs.zeek.org/en/master/scripts/policy/protocols/mqtt/main.zeek.html.  It looks like there are fields like ack,rec,rel,comp & qos_level that could be present.  If they could show up in the logs we should add them.

File: x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+- name: mqtt_subscribe
+  type: group
+  default_field: false
+  description: >
+    Fields exported by the Zeek mqtt_subscribe log.
+  fields:
+    - name: action
```
Comment (leehinman) [inline x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml:7]: should this also get copied to event.action?

File: x-pack/filebeat/module/zeek/mqtt_connect/_meta/fields.yml (hunk 1 above)
Comment (garanews) [reply x-pack/filebeat/module/zeek/mqtt_connect/_meta/fields.yml:20]: Yes, if are documented probably these fields can appear. so better to add. At that time when I tested, these fields didn't appear.

File: x-pack/filebeat/module/zeek/mqtt_publish/_meta/fields.yml (hunk 2 above)
Comment (garanews) [reply x-pack/filebeat/module/zeek/mqtt_publish/_meta/fields.yml:41]: Yes, if are documented probably these fields can appear. so better to add. At that time when I tested, these fields didn't appear.

File: x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml (hunk 3 above)
Comment (garanews) [reply x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml:7]: what do you mean?

Comment (garanews) [reply x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml:7]: @leehinman can you explain?

Comment (leehinman) [reply x-pack/filebeat/module/zeek/mqtt_subscribe/_meta/fields.yml:7]: Sorry I missed your earlier question.

event.action is an ECS [field](https://www.elastic.co/guide/en/ecs/current/ecs-event.html) that describes the action captured by the event.  It seems like this field is a good fit for that.  Much like id.orig_h maps to the ECS field source.address.  

We could either remove the custom field(zeek.mqtt_subscribe.action) and just use event.action or we could copy the value to event.action keeping the custom field (zeek.mqtt_subscribe.action).

**Pull request closed without merging**